service/history/workflow/mutable_state_impl.go

```diff
@@ -1544,12 +1544,8 @@ func (e *MutableStateImpl) ReplicateWorkflowExecutionStartedEvent(
 		e.executionInfo.ParentWorkflowId = event.ParentWorkflowExecution.GetWorkflowId()
 		e.executionInfo.ParentRunId = event.ParentWorkflowExecution.GetRunId()
 		e.executionInfo.ParentClock = parentClock
-	}
-
-	if event.ParentInitiatedEventId != 0 {
 		e.executionInfo.InitiatedId = event.GetParentInitiatedEventId()
-	} else {
-		e.executionInfo.InitiatedId = common.EmptyEventID
+		e.executionInfo.InitiatedVersion = event.GetParentInitiatedEventVersion()
 	}
```

Comment:
can we keep the else block?
```go
e.executionInfo.InitiatedId = common.EmptyEventID
e.executionInfo. InitiatedVersion = common.EmptyVersion
```

Reply:
Added back the check. also added a new if block to check the initiated version.